hoodie-common/src/main/java/com/uber/hoodie/common/table/log/HoodieLogFileReader.java

```diff
@@ -242,17 +264,19 @@ private boolean isBlockCorrupt(int blocksize) throws IOException {
     }
   }
 
-  private long scanForNextAvailableBlockOffset() throws IOException {
-    while (true) {
-      long currentPos = inputStream.getPos();
+  private long scanForNextAvailableBlockOffset(byte[] bytes, int numberOfByesRead)
+      throws IOException {
+    int offset = 0;
+    while (offset < numberOfByesRead) {
       try {
-        boolean isEOF = readMagic();
-        return isEOF ? inputStream.getPos() : currentPos;
-      } catch (CorruptedLogFileException e) {
+        readMagic(bytes, offset);
+        return offset;
+      } catch(CorruptedLogFileException io) {
         // No luck - advance and try again
-        inputStream.seek(currentPos + 1);
```

Comment:
would BufferedInputStream with mark and reset have helped resolve this issue differently?